docs/docs/python/goals/check.mdx

```diff
@@ -244,7 +244,7 @@ from mypy_boto3_ec2 import EC2Client
 
 if TYPE_CHECKING:
     # Safe, but will be flagged as a warning
```

Comment:
Thanks for taking the time to contribute. I think this code is already intended to be written along the lines of "you might need to add the pragma... or maybe use the better solution of having them as normal deps".


So, given that I think if we're making this change, we'll need to rewrite this comment, and also the text below:

> Alternatively, you can set `# pants: no-infer-dep` on the lines of type-stub imports "guarded" by a check of `if TYPE_CHECKING`.

An alternative to adding `pants: no-infer-dep` here would be leaving it as, but rephrasing the rest of the text if there's a way to make it clearer to you?

Reply:
Ah, I see your point—thanks for bringing that up! I’ll take another pass at clarifying this by editing the comment above.

It might be helpful to provide separate advice depending on whether `unowned_dependency_behavior` is set to `"error"` or `"warning"`. For our use case, we have `unowned_dependency_behavior` set to `"error"`, so it was absolutely necessary. Even with it set to `"warning"`, I feel it’s not ideal to have warnings consistently showing up. That could be especially confusing for users who struggle to determine whether a warning is safe to ignore or something that needs immediate attention. I’ll work on making that distinction clearer.

```toml
[python-infer]
unowned_dependency_behavior = "error"
```